tests/test_connection.py

```diff
@@ -3816,8 +3816,8 @@ def recv(self, maxlen):
             break
 
     # Check node_announcement has websocket
-    assert (only_one(l2.rpc.listnodes(l1.info['id'])['nodes'])['addresses']
-            == [{'type': 'ipv4', 'address': '127.0.0.1', 'port': port2}, {'type': 'websocket', 'port': ws_port}])
+    required_addresses = [{'type': 'ipv4', 'address': '127.0.0.1', 'port': port2}, {'type': 'websocket', 'port': ws_port}]
+    assert all(a in only_one(l2.rpc.listnodes(l1.info['id'])['nodes'])['addresses'] for a in required_addresses)
```

Comment:
```suggestion
    ws_address = {'type': 'websocket', 'port': ws_port}
    assert ws_address in only_one(l2.rpc.listnodes(l1.info['id'])['nodes'])['addresses'])
```

As the comment points out, we are checking only the web socket, so maybe we can discard to filter other types of address? without disable the IP discovery?

Reply:
Yep that would be cleaner. Also @endothermicdev the git message needs a `Changelog-None` as the last line for the PR to be mergable.

Reply:
Looks good to me.  I only left the ipv4 loopback in the check to preserve the previous functionality, but I agree it seems unnecessary.

Reply:
I added a new commit to remove the unmatched paren and insert the changelog.